preprocessing-service/preprocess.py

```diff
@@ -63,108 +63,112 @@ async def subscribe_handler(msg):
-                payload_data_df["filename"].notnull() & payload_data_df["filename"].str.contains(r"k3s\.log"),
-                ["is_control_plane_log", "kubernetes_component"],
-            ] = [True, "k3s"]
-            # rke2 kubelet
-            payload_data_df.loc[
-                payload_data_df["filename"].notnull() & payload_data_df["filename"].str.contains("rke2/agent/logs/kubelet"),
-                ["is_control_plane_log", "kubernetes_component"],
-            ] = [True, "kubelet"]
-        # k3s/rke2 systemd
-        if "COMM" in payload_data_df.columns:
-            comm_df_filter = payload_data_df["COMM"].notnull() & payload_data_df["COMM"].str.contains("(k3s|rke2)-(agent|server)|kubelet")
-            payload_data_df.loc[comm_df_filter, ["is_control_plane_log"]] = True
-            payload_data_df.loc[comm_df_filter, ["kubernetes_component"]] = payload_data_df[comm_df_filter]["COMM"].str.split("-").str[0]
-        # rke2/kops/kubeadm static pods
-        if "kubernetes.labels.tier" in payload_data_df.columns:
-            kube_labels_tier_filter = payload_data_df["kubernetes.labels.tier"].notnull() & payload_data_df["kubernetes.labels.tier"] == "control-plane"
-            payload_data_df.loc[kube_labels_tier_filter, ["is_control_plane_log"]] = True
-            payload_data_df.loc[kube_labels_tier_filter, ["kubernetes_component"]] = payload_data_df[kube_labels_tier_filter]["kubernetes.labels.component"]
+        payload_data_df["ingest_at"] = payload_data_df["ingest_at"].astype(str)
 
         masked_logs = []
+
         for index, row in payload_data_df.iterrows():
-            masked_logs.append(masker.mask(row["log"]))
+            try:
+                masked_logs.append(masker.mask(row["log"]))
+            except Exception as e:
+                masked_logs.append(row["log"])
         payload_data_df["masked_log"] = masked_logs
-        try:
-            async for ok, result in async_streaming_bulk(
-                es, doc_generator(payload_data_df)
-            ):
-                action, result = result.popitem()
-        except (BulkIndexError, ConnectionTimeout) as exception:
-            logging.error(exception)
-        is_control_log = payload_data_df["is_control_plane_log"] == True
-        control_plane_logs_df = payload_data_df[is_control_log]
-        '''
-        app_logs_df = payload_data_df[~is_control_log]
-        if len(app_logs_df) > 0:
-            await nw.publish("preprocessed_logs", app_logs_df.to_json().encode())
-        '''
-
-        if len(control_plane_logs_df) > 0:
+        payload_data_df.drop(["_type"], axis=1, errors="ignore", inplace=True)
+        payload_data_df.drop(["_version"], axis=1, errors="ignore", inplace=True)
+        opensearch_pending_list.append(payload_data_df)
+        opensearch_log_count += len(payload_data_df)
+        start_time = time.time()
+        if (start_time - last_time >= 1 and opensearch_log_count > 0) or opensearch_log_count >= 128:
```

Comment:
when is `opensearch_log_count` reset?

Reply:
That is part of an older version. Will remove this function altogether.